modules/@apostrophecms/schema/lib/addFieldTypes.js

```diff
@@ -1218,7 +1219,7 @@ module.exports = (self) => {
         self.validate(_field.schema, {
           type: 'relationship',
           subtype: _field.withType
-        });
+        }, _field);
```

Comment:
Passing the parent to `validate`. Should we modify `object` and `array` to use the same `validate` method instead of `validateField`. (the validate method keeps tracks of already validated fields). I think it would make sense to use the same method for `relationship`, `object` and `array`.